infrastructure/users_stack.py

```diff
@@ -11,21 +11,25 @@ class UsersStack(Stack):
     def __init__(self, scope: Construct, stack_id: str, **kwargs: Any) -> None:
         super().__init__(scope, stack_id, **kwargs)
 
-        account_ids = (
-            aws_iam.AccountPrincipal(account_id=account_id)
-            for account_id in os.environ.get(
-                "DATALAKE_USERS_AWS_ACCOUNTS_IDS", aws_iam.AccountRootPrincipal().account_id
-            ).split(",")
-        )
+        saml_provider_arn = os.environ.get("DATALAKE_SAML_IDENTITY_PROVIDER_ARN", None)
+
+        if saml_provider_arn:
+            principal = aws_iam.FederatedPrincipal(
+                federated=saml_provider_arn,
+                assume_role_action="sts:AssumeRoleWithSAML",
+                conditions={"StringEquals": {"SAML:aud": "https://signin.aws.amazon.com/saml"}},
+            )
 
-        principals = aws_iam.CompositePrincipal(*account_ids)
+        else:
+            principal = aws_iam.AccountPrincipal(  # type: ignore[assignment]
+                account_id=aws_iam.AccountRootPrincipal().account_id
+            )
 
         self.users_role = aws_iam.Role(
             self,
             "users-role",
             role_name=ResourceName.USERS_ROLE_NAME.value,
-            assumed_by=principals,  # type: ignore[arg-type]
+            assumed_by=principal,  # type: ignore[arg-type]
```

Comment:
Is the `# type: ignore[arg-type]` annotation superfluous now?

Reply:
No, when removed I get following error:

```
infrastructure/users_stack.py:32: error: Argument "assumed_by" to "Role" has incompatible type "FederatedPrincipal"; expected "IPrincipal"  [arg-type]
infrastructure/users_stack.py:32: note: Following member(s) of "FederatedPrincipal" have conflicts:
infrastructure/users_stack.py:32: note:     Expected:
infrastructure/users_stack.py:32: note:         def __jsii_proxy_class__() -> Type[_IPrincipalProxy]
infrastructure/users_stack.py:32: note:     Got:
infrastructure/users_stack.py:32: note:         def __jsii_proxy_class__() -> Type[_PrincipalBaseProxy]

```